File: components/redis-handling-session/src/main/java/com/sv/youapp/component/authorization/entities/redis/OAuth2ClientCredentialsGrantAuthorization.java
package com.sv.youapp.component.authorization.entities.redis;

public class OAuth2ClientCredentialsGrantAuthorization extends AbstractGrantAuthorization {

	public OAuth2ClientCredentialsGrantAuthorization(
		String id,
		String registeredClientId,
		String principalName,
		String scopes,
		AbstractGrantAuthorization.AccessToken accessToken
	) {
		super(id, registeredClientId, principalName, scopes, accessToken, null);
	}
}

File: platform-instances/authorization-server/src/main/kotlin/com/sv/youapp/authorization/entities/redis/OAuth2ClientCredentialsGrantAuthorization.kt
package com.sv.youapp.authorization.entities.redis

class OAuth2ClientCredentialsGrantAuthorization(
    id: String,
    registeredClientId: String,
    principalName: String,
    scopes: String,
    accessToken: AccessToken?,
) : AbstractGrantAuthorization(id, registeredClientId, principalName, scopes, accessToken, null)
